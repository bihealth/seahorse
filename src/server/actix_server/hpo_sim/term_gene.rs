//! Entry point `/hpo/sim/term-gene` that allows the similarity computation between a set of
//! terms and a gene.

use std::str::FromStr;

use actix_web::{
    get,
    web::{self, Data, Json, Path},
    Responder,
};

use hpo::{annotations::GeneId, term::HpoGroup, HpoTermId, Ontology};

use super::super::CustomError;
use crate::{query, server::WebServerData};

/// Enum for representing similarity method to use.
#[derive(Default, Debug, Clone, Copy, derive_more::Display)]
pub enum SimilarityMethod {
    /// Phenomizer similarity score.
    #[default]
    #[display(fmt = "phenomizer")]
    Phenomizer,
}

impl FromStr for SimilarityMethod {
    type Err = anyhow::Error;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        Ok(match s {
            "resnik::gene" => Self::Phenomizer,
            _ => anyhow::bail!("unknown similarity method: {}", s),
        })
    }
}

/// Parameters for `handle`.
///
/// This allows to compute differences between
///
/// - `terms` -- set of terms to use as query
/// - `gene_ids` -- set of ids for genes to use as "database"
/// - `gene_symbols` -- set of symbols for genes to use as
///   "database"
#[derive(serde::Deserialize, Debug, Clone)]
struct Request {
    /// Set of terms to use as query.
    #[serde(deserialize_with = "super::super::vec_str_deserialize")]
    pub terms: Vec<String>,
    /// The set of ids for genes to use as "database".
    #[serde(
        default = "Option::default",
        skip_serializing_if = "Option::is_none",
        deserialize_with = "super::super::option_vec_str_deserialize"
    )]
    pub gene_ids: Option<Vec<String>>,
    /// The set of symbols for genes to use as "database".
    #[serde(
        default = "Option::default",
        skip_serializing_if = "Option::is_none",
        deserialize_with = "super::super::option_vec_str_deserialize"
    )]
    pub gene_symbols: Option<Vec<String>>,
    /// The similarity method to use.
    #[serde(
        deserialize_with = "help::similarity_deserialize",
        default = "help::default_sim"
    )]
    pub sim: SimilarityMethod,
}

/// Helpers for deserializing `Request`.
mod help {
    /// Helper to deserialize a similarity method.
    pub fn similarity_deserialize<'de, D>(
        deserializer: D,
    ) -> Result<super::SimilarityMethod, D::Error>
    where
        D: serde::Deserializer<'de>,
    {
        let s = <String as serde::Deserialize>::deserialize(deserializer)?;
        std::str::FromStr::from_str(&s).map_err(serde::de::Error::custom)
    }

    /// Default value for `Request::sim`.
    pub fn default_sim() -> super::SimilarityMethod {
        super::SimilarityMethod::Phenomizer
    }
}

/// Query for similarity between a set of terms to each entry in a
/// list of genes.
#[allow(clippy::unused_async)]
#[get("/hpo/sim/term-gene")]
async fn handle(
    data: Data<WebServerData>,
    _path: Path<()>,
    query: web::Query<Request>,
) -> actix_web::Result<impl Responder, CustomError> {
    let _ = &query.sim;

    let hpo: &Ontology = &data.ontology;

    // Translate strings from the query into an `HpoGroup`.
    let query_terms = {
        let mut query_terms = HpoGroup::new();
        for term in &query.terms {
            if let Some(term) = hpo.hpo(HpoTermId::from(term.clone())) {
                query_terms.insert(term.id());
            }
        }
        query_terms
    };

    // Translate strings from the query into genes via symbol or gene ID.
    let genes = if let Some(gene_ids) = &query.gene_ids {
        Ok(gene_ids
            .iter()
            .filter_map(|gene_id| {
                if let Ok(gene_id) = gene_id.parse::<u32>() {
                    hpo.gene(&GeneId::from(gene_id))
                } else {
                    None
                }
            })
            .collect::<Vec<_>>())
    } else if let Some(gene_symbols) = &query.gene_symbols {
        Ok(gene_symbols
            .iter()
            .filter_map(|gene_symbol| hpo.gene_by_name(gene_symbol))
            .collect::<Vec<_>>())
    } else {
        Err(CustomError::new(anyhow::anyhow!(
            "either `gene_ids` or `gene_symbols` must be given"
        )))
    }?;

    // Perform similarity computation.
    let result = query::run_query(
        &query_terms,
        &genes,
        hpo,
        data.db.as_ref().expect("must provide RocksDB"),
    )
    .map_err(CustomError::new)?;

    Ok(Json(result))
}

#[cfg(test)]
mod test {
    /// Helper function for running a query.
    #[allow(dead_code)]
    async fn run_query(uri: &str) -> Result<crate::query::query_result::Container, anyhow::Error> {
        let hpo_path = "tests/data/hpo";
        let ontology = crate::common::load_hpo(hpo_path)?;
        let db = Some(rocksdb::DB::open_cf_for_read_only(
            &rocksdb::Options::default(),
            format!("{}/{}", hpo_path, "resnik"),
            ["meta", "resnik_pvalues"],
            true,
        )?);

        let app = actix_web::test::init_service(
            actix_web::App::new()
                .app_data(actix_web::web::Data::new(crate::server::WebServerData {
                    ontology,
                    db,
                }))
                .service(super::handle),
        )
        .await;
        let req = actix_web::test::TestRequest::get().uri(uri).to_request();
        let resp: crate::query::query_result::Container =
            actix_web::test::call_and_read_body_json(&app, req).await;

        Ok(resp)
    }

    #[actix_web::test]
    async fn hpo_sim_term_gene_terms_gene_ids() -> Result<(), anyhow::Error> {
        Ok(insta::assert_yaml_snapshot!(
            &run_query("/hpo/sim/term-gene?terms=HP:0010442,HP:0000347&gene_ids=23483,7273&sim=resnik::gene")
                .await?
        ))
    }

    #[actix_web::test]
    async fn hpo_sim_term_gene_terms_symbols() -> Result<(), anyhow::Error> {
        Ok(insta::assert_yaml_snapshot!(
            &run_query("/hpo/sim/term-gene?terms=HP:0010442,HP:0000347&gene_symbols=TGDS,TTN&sim=resnik::gene")
                .await?
        ))
    }
}
