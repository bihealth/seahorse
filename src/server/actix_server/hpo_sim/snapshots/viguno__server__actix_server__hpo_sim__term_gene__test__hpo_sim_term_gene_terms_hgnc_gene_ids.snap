---
source: src/server/actix_server/hpo_sim/term_gene.rs
expression: "&run_query(\"/hpo/sim/term-gene?terms=HP:0010442,HP:0000347&gene_ids=HGNC:20324,HGNC:12403\").await?"
---
version:
  hpo: 2023-04-05
  viguno: 0.0.0
query:
  terms:
    - term_id: "HP:0000347"
      term_name: Micrognathia
    - term_id: "HP:0010442"
      term_name: Polydactyly
  genes:
    - entrez_id: 7273
      gene_symbol: TTN
      hgnc_id: "HGNC:12403"
    - entrez_id: 23483
      gene_symbol: TGDS
      hgnc_id: "HGNC:20324"
result:
  - gene_symbol: TGDS
    p_value: 0.1
    score: 10
    terms:
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000347"
          term_name: Micrognathia
        score: 1.8309557
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0001831"
          term_name: Short toe
        score: 1.4382622
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0003097"
          term_name: Short femur
        score: 1.4382622
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0005792"
          term_name: Short humerus
        score: 1.4382622
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0010049"
          term_name: Short metacarpal
        score: 1.4382622
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000272"
          term_name: Malar flattening
        score: 1.341162
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001181"
          term_name: Adducted thumb
        score: 1.2074362
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0004209"
          term_name: Clinodactyly of the 5th finger
        score: 1.2074362
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0009464"
          term_name: Ulnar deviation of the 2nd finger
        score: 1.2074362
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0009467"
          term_name: Radial deviation of the 2nd finger
        score: 1.2074362
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0030368"
          term_name: Hyperphalangy of the 2nd finger
        score: 1.2074362
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0100490"
          term_name: Camptodactyly of finger
        score: 1.2074362
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0002857"
          term_name: Genu valgum
        score: 1.0733097
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0010508"
          term_name: Metatarsus valgus
        score: 1.0733097
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0005930"
          term_name: Abnormal epiphysis morphology
        score: 1.002138
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0000954"
          term_name: Single transverse palmar crease
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001762"
          term_name: Talipes equinovarus
        score: 0.73446083
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000470"
          term_name: Short neck
        score: 0.6371871
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000767"
          term_name: Pectus excavatum
        score: 0.6371871
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000768"
          term_name: Pectus carinatum
        score: 0.6371871
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0002650"
          term_name: Scoliosis
        score: 0.6371871
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001373"
          term_name: Joint dislocation
        score: 0.48051217
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001387"
          term_name: Joint stiffness
        score: 0.44339323
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001388"
          term_name: Joint laxity
        score: 0.44339323
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0005692"
          term_name: Joint hyperflexibility
        score: 0.44339323
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000160"
          term_name: Narrow mouth
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000162"
          term_name: Glossoptosis
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000175"
          term_name: Cleft palate
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000193"
          term_name: Bifid uvula
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000201"
          term_name: Pierre-Robin sequence
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000204"
          term_name: Cleft upper lip
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000218"
          term_name: High palate
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000293"
          term_name: Full cheeks
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000460"
          term_name: Narrow nose
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000579"
          term_name: Nasolacrimal duct obstruction
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000582"
          term_name: Upslanted palpebral fissure
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0002553"
          term_name: Highly arched eyebrow
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0009933"
          term_name: Narrow naris
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0010285"
          term_name: Oral synechia
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0010763"
          term_name: Low insertion of columella
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0045074"
          term_name: Thin eyebrow
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000476"
          term_name: Cystic hygroma
        score: 0.4094388
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0000023"
          term_name: Inguinal hernia
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001537"
          term_name: Umbilical hernia
        score: 0.28058913
      - term_query: ~
        term_gene:
          term_id: "HP:0000028"
          term_name: Cryptorchidism
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0000316"
          term_name: Hypertelorism
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0000368"
          term_name: "Low-set, posteriorly rotated ears"
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0000389"
          term_name: Chronic otitis media
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0000520"
          term_name: Proptosis
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001250"
          term_name: Seizure
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001263"
          term_name: Global developmental delay
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001508"
          term_name: Failure to thrive
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001511"
          term_name: Intrauterine growth retardation
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001629"
          term_name: Ventricular septal defect
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001631"
          term_name: Atrial septal defect
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001651"
          term_name: Dextrocardia
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001680"
          term_name: Coarctation of aorta
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002119"
          term_name: Ventriculomegaly
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002623"
          term_name: Overriding aorta
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0004322"
          term_name: Short stature
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0008897"
          term_name: Postnatal growth retardation
        score: 0
  - gene_symbol: TTN
    p_value: 0.3
    score: 5.2287874
    terms:
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000308"
          term_name: Microretrognathia
        score: 1.8309557
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000303"
          term_name: Mandibular prognathia
        score: 1.5641111
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0100807"
          term_name: Long fingers
        score: 1.2074362
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001385"
          term_name: Hip dysplasia
        score: 1.002138
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003273"
          term_name: Hip contracture
        score: 1.002138
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0000982"
          term_name: Palmoplantar keratoderma
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001761"
          term_name: Pes cavus
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001762"
          term_name: Talipes equinovarus
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001763"
          term_name: Pes planus
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001771"
          term_name: Achilles tendon contracture
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003547"
          term_name: Shoulder girdle muscle weakness
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003691"
          term_name: Scapular winging
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003731"
          term_name: Quadriceps muscle weakness
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003749"
          term_name: Pelvic girdle muscle weakness
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0008959"
          term_name: Distal upper limb muscle weakness
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0008963"
          term_name: Tibialis muscle weakness
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0008981"
          term_name: Calf muscle hypertrophy
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0009027"
          term_name: Foot dorsiflexor weakness
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0009049"
          term_name: Peroneal muscle atrophy
        score: 0.73446083
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0009077"
          term_name: Weakness of long finger extensor muscles
        score: 0.73446083
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0002650"
          term_name: Scoliosis
        score: 0.6371871
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0003306"
          term_name: Spinal rigidity
        score: 0.6371871
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0003307"
          term_name: Hyperlordosis
        score: 0.6371871
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0002828"
          term_name: Multiple joint contractures
        score: 0.48051217
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0008800"
          term_name: Limited hip movement
        score: 0.44339323
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000160"
          term_name: Narrow mouth
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000193"
          term_name: Bifid uvula
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000218"
          term_name: High palate
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0000276"
          term_name: Long face
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0001349"
          term_name: Facial diplegia
        score: 0.4183606
      - term_query:
          term_id: "HP:0000347"
          term_name: Micrognathia
        term_gene:
          term_id: "HP:0005991"
          term_name: Limited neck flexion
        score: 0.4094388
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0001290"
          term_name: Generalized hypotonia
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0002355"
          term_name: Difficulty walking
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0002359"
          term_name: Frequent falls
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0002421"
          term_name: Poor head control
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003323"
          term_name: Progressive muscle weakness
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003324"
          term_name: Generalized muscle weakness
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003327"
          term_name: Axial muscle weakness
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003391"
          term_name: Gowers sign
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003403"
          term_name: "EMG: decremental response of compound muscle action potential to repetitive nerve stimulation"
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003458"
          term_name: "EMG: myopathic abnormalities"
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003551"
          term_name: Difficulty climbing stairs
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003555"
          term_name: Muscle fiber splitting
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003557"
          term_name: Increased variability in muscle fiber diameter
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003687"
          term_name: Centrally nucleated skeletal muscle fibers
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003700"
          term_name: Generalized amyotrophy
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003715"
          term_name: Myofibrillar myopathy
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003722"
          term_name: Neck flexor weakness
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003741"
          term_name: Congenital muscular dystrophy
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003789"
          term_name: Minicore myopathy
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003803"
          term_name: Type 1 muscle fiber predominance
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0003805"
          term_name: Rimmed vacuoles
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0004889"
          term_name: Intermittent episodes of respiratory insufficiency due to muscle weakness
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0008978"
          term_name: Necrotizing myopathy
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0009046"
          term_name: Difficulty running
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0009058"
          term_name: Increased muscle lipid content
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0009113"
          term_name: Diaphragmatic weakness
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0012548"
          term_name: Fatty replacement of skeletal muscle
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0030091"
          term_name: Absent muscle fiber merosin
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0031237"
          term_name: Internally nucleated skeletal muscle fibers
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0031374"
          term_name: Ankle weakness
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0100293"
          term_name: Hypertrophied muscle fibers
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0100295"
          term_name: Muscle fiber atrophy
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0100297"
          term_name: Increased endomysial connective tissue
        score: 0.28058913
      - term_query:
          term_id: "HP:0010442"
          term_name: Polydactyly
        term_gene:
          term_id: "HP:0100578"
          term_name: Lipoatrophy
        score: 0.28058913
      - term_query: ~
        term_gene:
          term_id: "HP:0000407"
          term_name: Sensorineural hearing impairment
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0000411"
          term_name: Protruding ear
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0000508"
          term_name: Ptosis
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0000602"
          term_name: Ophthalmoplegia
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0000750"
          term_name: Delayed speech and language development
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001256"
          term_name: "Intellectual disability, mild"
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001260"
          term_name: Dysarthria
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001284"
          term_name: Areflexia
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001508"
          term_name: Failure to thrive
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001618"
          term_name: Dysphonia
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001620"
          term_name: High pitched voice
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001634"
          term_name: Mitral valve prolapse
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001635"
          term_name: Congestive heart failure
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001639"
          term_name: Hypertrophic cardiomyopathy
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001644"
          term_name: Dilated cardiomyopathy
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001645"
          term_name: Sudden cardiac death
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001667"
          term_name: Right ventricular hypertrophy
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001678"
          term_name: Atrioventricular block
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001708"
          term_name: Right ventricular failure
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001712"
          term_name: Left ventricular hypertrophy
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0001874"
          term_name: Abnormality of neutrophils
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002091"
          term_name: Restrictive ventilatory defect
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002194"
          term_name: Delayed gross motor development
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002312"
          term_name: Clumsiness
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002515"
          term_name: Waddling gait
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002527"
          term_name: Falls
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002792"
          term_name: Reduced vital capacity
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002877"
          term_name: Nocturnal hypoventilation
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0002878"
          term_name: Respiratory failure
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0003376"
          term_name: Steppage gait
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0004322"
          term_name: Short stature
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0004756"
          term_name: Ventricular tachycardia
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0005110"
          term_name: Atrial fibrillation
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0006699"
          term_name: Premature atrial contractions
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0008180"
          term_name: Mildly elevated creatine kinase
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0011717"
          term_name: Atrioventricular reentrant tachycardia
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0012666"
          term_name: Severely reduced left ventricular ejection fraction
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0012764"
          term_name: Orthopnea
        score: 0
      - term_query: ~
        term_gene:
          term_id: "HP:0030059"
          term_name: Mitochondrial depletion
        score: 0

